airflow/api_connexion/endpoints/dag_run_endpoint.py

```diff
@@ -204,6 +207,10 @@ def get_dag_runs_batch(*, session: Session = NEW_SESSION) -> APIResponse:
     else:
         query = query.filter(DagRun.dag_id.in_(readable_dag_ids))
 
+    if data.get("states"):
+        states = set(data["states"])
+        query = query.filter(DagRun.state.in_(states))
```

Comment:
```suggestion
    states = data.get("states")
    if states:
        query = query.filter(DagRun.state.in_(states))
```

Python variables are function-scoped, not block-scoped, so it’s not useful to put the variable inside the `if` block. And moving the variable out avoids one unnecessary member access overhead.

Reply:
Hi @uranusjr , thanks for your recommendation, but after that change I had this errors:
E       TypeError: 'NoneType' object is not iterable

So I put it back as the other lines (f.i. variable dag_ids, line 205) with the variable inside the block.